src/encoded/static/components/search.js

```diff
@@ -700,12 +705,94 @@ TextFilter.propTypes = {
 // Displays the entire list of facets. It contains a number of <Facet> components.
 export const FacetList = (props) => {
     const { context, facets, filters, mode, orientation, hideTextFilter, addClasses, docTypeTitleSuffix, supressTitle, onFilter } = props;
+
+    const [expandedFacets, setExpandFacets] = React.useState(new Set());
+
+    React.useEffect(() => {
+        // Get facets from storage that need to be expanded
+        const facetsStorage = sessionStorage.getItem(FACET_STORAGE);
+        const facetList = new Set(facetsStorage ? facetsStorage.split(',') : []);
+
+
+        sessionStorage.setItem(FACET_STORAGE, facetList.size !== 0 ? [...facetList].join(',') : []);
+        setExpandFacets(facetList); // initalize facet collapse-state
+    }, []);
```

Comment:
I think "setFieldAsNotNewlyLoaded" needs to be commented better and it not properly named. When the page first loads, any facets with a selected item should be opened. For example 

http://localhost:6543/search/?type=Experiment&award.project=ENCODE

should have project opened. However, it should not be opened on subsequent clicks. "setFieldAsNotNewlyLoaded" helps with that as it is located somewhere where it can be called once. I placed facet.open_on_load check in this method too because placing it on the consuming affect resulted in it being called multiple times. Please, let me know if my idea makes no sense.

Reply:
What do you mean by “should not be opened on subsequent clicks”?

Reply:
Great tip. The code was refactored and now is simplified.